node/router/src/helpers/peer.rs

```diff
@@ -82,9 +83,14 @@ impl<N: Network> Peer<N> {
     }
 
     /// Promote a connecting peer to a fully connected one.
-    pub fn upgrade_to_connected(&mut self, connected_addr: SocketAddr, cr: &ChallengeRequest<N>, router: Router<N>) {
+    pub fn upgrade_to_connected(
+        &mut self,
+        connected_addr: SocketAddr,
+        cr: &ChallengeRequest<N>,
+        router: Router<N>,
+    ) -> Result<()> {
         // Logic check: this can only happen during the handshake.
-        assert!(matches!(self, Self::Connecting(_)));
+        ensure!(matches!(self, Self::Connecting(_)), "Peer is not in `connecting` state");
```

Comment:
It seems to me that this is the only change relevant to the PR title. 

Are the log changes related/necessary?

Reply:
I ended up adding a lot of logging to this PR while we were debugging testnet issues. Sorry for the confusing.